### PR TITLE
Improve wording of warning message.

### PR DESCRIPTION
**Pull request by adrian-prantl**

### CODE REVIEW

**Comment by adrian-prantl** [comment]
@swift-ci test

File: source/Target/Target.cpp [hunk 1]
```diff
@@ -2158,17 +2158,17 @@ TypeSystem *Target::GetScratchTypeSystemForLanguage(
                                     .GetFilename()
                                     .AsCString();
             if (m_use_scratch_typesystem_per_module)
-              errs->Printf(
-                  "\nnote: Swift header search options for %s conflict with "
-                  "options found in other modules. Symbols from other modules "
-                  "may not be available.",
-                  module_name);
-            else
+              errs->Printf("\nnote: Swift ClangImporter options for %s "
```
Comment (fredriss) [inline source/Target/Target.cpp]: Is ClangImporter really something we want to expose? Wouldn't it be better to just say "compiler options"?

**Comment by adrian-prantl** [comment]
@swift-ci test and merge

File: source/Target/Target.cpp (hunk 1 above)
Comment (adrian-prantl) [reply source/Target/Target.cpp]: It might actually be more accurate, too.

File: source/Target/Target.cpp [hunk 2]
```diff
@@ -2158,17 +2158,17 @@ TypeSystem *Target::GetScratchTypeSystemForLanguage(
...
-              errs->Printf(
-                  "\nnote: Swift header search options for %s conflict with "
-                  "options found in other modules. Symbols from other modules "
-                  "may not be available.",
-                  module_name);
-            else
+              errs->Printf("\nnote: Swift compiler options for %s "
+                           "conflict with options found in other modules;\n"
+                           "      Switching to an isolated expression context "
+                           "for %s.\n", module_name, module_name);
```
Comment (fredriss) [inline source/Target/Target.cpp]: I'm also wondering if this is informative enough. I think we should add something like:
"..isolated expression context for %s, you won't have access to old $R variables anymore."

Comment (adrian-prantl) [reply source/Target/Target.cpp]: I wanted to strike a balance between an informative and concise message. That's why I removed the message in the curly braces below from this path. My thinking was that "isolated" implies this. If you think it's important enough to include I can back that change out again.

Comment (fredriss) [reply source/Target/Target.cpp]: I don't think people will understand what isolated means here. Maybe not even expression context. What about the less accurate but (I think more understandable):
"Switching to a new expression evaluator for %s, old $R variables are lost."

Comment (fredriss) [reply source/Target/Target.cpp]: or "are not accessible anymore"

**Comment by adrian-prantl** [comment]
@swift-ci test and merge

**Pull request merged**